### PR TITLE
chore: update all deps

### PR DESCRIPTION
**Pull request by sethvargo**
Closes https://github.com/google-github-actions/upload-cloud-storage/pull/275

- https://github.com/vercel/ncc/pull/921 is preventing us from updating typescript and a few other deps.
- https://github.com/googleapis/nodejs-storage/issues/1968 is preventing us from upgrading cloud-storage.

### CODE REVIEW

**Comment by sethvargo** [comment]
Weird - looks like a typescript bug?

**Comment by verbanicm** [comment]
run `npm update` and it should work

**Comment by sethvargo** [comment]
I did - I even deleted node_modules, and it's still failing with the same error locally.

File: package.json [hunk 1]
```diff
@@ -23,28 +23,28 @@
...
-    "eslint": "^8.14.0",
+    "eslint": "^8.16.0",
     "mocha": "^10.0.0",
     "prettier": "^2.6.2",
-    "sinon": "^13.0.2",
-    "ts-node": "^10.7.0",
-    "typescript": "^4.6.4"
+    "sinon": "^14.0.0",
+    "ts-node": "^10.8.0",
+    "typescript": "^4.6.3"
```
Comment (verbanicm) [inline package.json:48]: Why did this go backwards?

**Review by verbanicm — APPROVED** [review]

**Pull request merged**